### PR TITLE
libxml2: fix FreeBSD

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **libxml2/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
build passed on https://github.com/ericLemanissier/conan-center-index/runs/1578497224

**Review by Croydon — APPROVED** [review]

File: recipes/libxml2/all/conanfile.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def package_info(self):
             self.output.info("Appending PATH environment variable: {}".format(bindir))
             self.env_info.PATH.append(bindir)
 
-        if self.settings.os == "Linux" or self.settings.os == "Macos":
+        if self.settings.os in ["Linux", "Macos", "FreeBSD"]:
```
Comment (theirix) [inline recipes/libxml2/all/conanfile.py:258]: Maybe just put it in `else` block for a `if self.settings.os == "Windows"`?

Comment (madebr) [reply recipes/libxml2/all/conanfile.py:258]: I"m not sure, but `m` can also be used with `mingw`.
So testing for Windows is not enough.

Comment (theirix) [reply recipes/libxml2/all/conanfile.py:258]: That's correct. Mingw makes life harder.

Comment (ericLemanissier) [reply recipes/libxml2/all/conanfile.py:258]: I'm not sure either, it's usually better to be explicit in this case

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**